Add awesome list of software tools and resources

santisoler opened the pull request:
Add a new Markdown file with an awesome list of software tools and resources.

santisoler commented:
@leouieda @indiauppal I created a small Markdown file with some tools that might be interesting for anyone in the lab. The file lives in the root of this repo, but feel free to propose another home for it.

leouieda commented:
Thanks @santisoler! Great stuff and I already opened some of those links to read later 🙂 

How about renaming the file to just `resources.md` to make the URL a bit shorter?

@indiauppal feel free to comment and make suggestions before we merge this (look at the "Files changed" tab where you can leave comments on individual lines). Also feel free to add things to the list later after it's merged.

indiauppal commented on awesome-software-and-resources.md:
```diff
@@ -0,0 +1,142 @@
+Every one of the software listed on below are available through [open-source or
+Software Libre](https://en.wikipedia.org/wiki/Free_software) licenses.
+
+- LaTeX and document writing
```
[Overleaf] (https://www.overleaf.com/): a LaTeX editor with lots of templates. Useful for UG dissertation students.

santisoler replied:
Absolutely! We are used to write our papers in GitHub repos, but for anyone that isn't used to the GitHub workflow and, on top of that, must learn LaTeX, Overleaf is a great alternative. Thanks @indiauppal!

BTW, you can make suggestions while reviewing files. You just need to click the first icon in the comment (the one that looks like a paper sheet with a plus and a minus). That creates a commit suggestion that can be accepted right away with a single click. Try out next time!

indiauppal commented:
Thank you @santisoler for creating this!!!
I'm happy for you to merge. As I come across useful resources I'll add them.
I've made a comment about Overleaf - a LaTeX editor. It has lots of templates, the UG dissertation students might find it useful because they have less time to learn LaTeX.

santisoler commented:
I've just renamed the file to `resources.md` after @leouieda suggestions.
I've added Overleaf, as @indiauppal pointed out, but moved it under "Online services".
I've also added the guide for writing commit messages that @SAskevold shared on Slack.

leouieda reviewed (approved):
🎉  Merging!
